### ISSUE TITLE
wrong commit message

### ISSUE THREAD

**Issue by doughsay** [dependabot/feedback#660]
Hi,

Today all my dependency updates have the same commit message; it seems it somehow got "stuck". See this example screenshot:

<img width="576" alt="Screen Shot 2019-09-03 at 3 27 15 PM" src="https://user-images.githubusercontent.com/221693/64212819-6dbf9d80-ce5f-11e9-9d16-0e6bd849bdf0.png">

The PR title and description are correct, it's just the commit description that's wrong.
I've merged a bunch together and they're all the same:

<img width="706" alt="Screen Shot 2019-09-03 at 3 28 44 PM" src="https://user-images.githubusercontent.com/221693/64212840-8b8d0280-ce5f-11e9-80b5-078bc1245882.png">

**Comment by greysteil** [comment]
That is *really* weird. We don't have  any caching around commit message generation  so I can't think of any way that would happen on our side. Is the repo public? And do you see those messages when you run `git log` locally?

**Comment by doughsay** [comment]
Yep, really weird indeed!  It's definitely in the actual commit message, here's my local commit log:

<img width="752" alt="Screen Shot 2019-09-03 at 5 53 12 PM" src="https://user-images.githubusercontent.com/221693/64217839-cf8a0280-ce73-11e9-82e6-54de2ac88adb.png">

And here's a `git show` on one of those commits:

<img width="1141" alt="Screen Shot 2019-09-03 at 5 54 45 PM" src="https://user-images.githubusercontent.com/221693/64217872-f3e5df00-ce73-11e9-8cb4-f5fb45ae817e.png">

Sorry the repo isn't public...  I have several public repos using Dependabot, but none of them show this issue.  This is only happening on one private repo.  Maybe it'll work itself out; I'll keep an eye on it and see if it happens the next time a bump comes in.

And just to throw in some more info: as you can see by the timestamps, it was me quickly merging many PRs, after they had all been recently rebased.

**Comment by greysteil** [comment]
The culprit here is likely to be [this method](https://github.com/dependabot/dependabot-core/blob/master/common/lib/dependabot/pull_request_updater/github.rb#L170-L189), but the lookup logic there is relatively robust. What process did you use to merge the commits? Did you merge them into each other or anything similar?

I've never seen or heard of this happen before.

**Comment by doughsay** [comment]
So I may have done some weird things leading up to this point, let me see if I can describe the process:

1.) Dependabot opens all PRs against a branch called `dependency-updates`, which is occasionally merged into master and then updated to match master.
2.) I had many (~ 10) open dependency PRs waiting
3.) I rebased `dependency-updates` onto another branch containing a large refactor of our codebase
4.) dependabot saw that the base branch of all the update PRs changed, so dependant started rebasing all the PRs (this took a while)
5.) once all the PRs were rebased and CI passed, I started quickly clicking the merge buttons in the GitHub UI for each PR, not giving them a chance to each kick off another rebase round.

That's when I noticed the commit messages were all the same.

We had another dependency update come in this morning, and the message is **correct**, it is not broken now.  I'm not that upset by all the wrong messages, I just thought I'd let you know it happened in case something could come of it.  For now things seem to have gone back to normal.

Thanks for your help and time!
